src/py21cmmc/likelihood.py

```diff
@@ -682,31 +684,283 @@ def store(self, model, storage):
             storage.update({k + "_%s" % i: v for k, v in m.items()})
 
 
+class LikelihoodPlanckPowerSpectra(LikelihoodBase):
+    r"""A likelihood template to use Planck power spectrum.
+
+    It makes use of the clik code developed by the planck collaboration.
+    Go to the planck legacy archive website to download the code and data and get information: https://pla.esac.esa.int
+    You should download the "baseline" data package.
+    This likelihood currently supports Planck_lensing, Planck_highl_TTTEEE and Planck_lowl_EE data. Could easily be extended if required.
+
+    Parameters
+    ----------
+    datafolder : str
+        folder that contains Planck data if "clik" format.
+    name_lkl : str
+        default = Planck_lowl_EE, the planck likelihood to compute. choice: Planck_lensing, Planck_highl_TTTEEE, Planck_lowl_EE
+    """
+
+    required_cores = ((core.CoreLightConeModule, core.CoreCMB),)
+
+    def __init__(
+        self,
+        *args,
+        datafolder="./",
```

Comment:
I'm wondering if we need to provide the Planck likelihood packages, which can be pretty large if we put them in the `data` folder (e.g. 80MB for Planck 2018 and >200MB for 2015). `datafolder` here refers to the folder where these likelihood codes are stored. when different `name_lkl` is used, users need to change `datafolder` as well to the folder where `name_lkl` is stored. The original Planck likelihood packages adopt folder names which are not very comprehensive (e.g.  `low_l/simall/simall_100x143_offlike5_EE_Aplanck_B.clik`)

I guess if we want to provide plik along with py21cmmc, we need to alter the folder names accordingly, so that it's easy for users who are not familiar with plik to use this likelihood

Reply:
However, without plik, I cannot add test for this likelihood...

Reply:
I *think* it would be best to have an automatic downloader that gets the plik likelihoods when requested, and puts them somewhere in the user's home folder or something. This might make testing a bit annoying, but I think we can auto-cache it on travis as well.